core/src/main/java/hudson/FilePath.java

```diff
@@ -502,6 +522,7 @@ public Integer invoke(File f, VirtualChannel channel) throws IOException {
                 } finally {
                     a.close();
                 }
+                function.accept(a.files());
```

Comment:
this looks like it is called *after* the fact.  if somone wants logging of the operations then surely the want them as it happens.
for example if it takes 1 minute to archive/unarchive something then I would as a suser expect to see what was beeing archvied as it happened not nothing for 1 minute and then a sprew of everything.